// @vibe/shared/dist/tests/constants.js
export var ComponentDefaultTestId;
(function (ComponentDefaultTestId) {
    // Don't remove next line
    // plop_marker:default-data-testid-declarations
    ComponentDefaultTestId["EMPTY_STATE"] = "empty-state";
    ComponentDefaultTestId["TRANSITION_VIEW"] = "transition-view";
    ComponentDefaultTestId["TEXT_AREA"] = "text-area";
    ComponentDefaultTestId["EDITABLE_TEXT"] = "editable-text";
    ComponentDefaultTestId["TIPSEEN_MEDIA"] = "tipseen-media";
    ComponentDefaultTestId["INDICATOR"] = "indicator";
    ComponentDefaultTestId["BADGE"] = "badge";
    ComponentDefaultTestId["TITLE"] = "title";
    ComponentDefaultTestId["TEXT"] = "text";
    ComponentDefaultTestId["COMBOBOX"] = "combobox";
    ComponentDefaultTestId["COMBOBOX_CATEGORY"] = "combobox-category";
    ComponentDefaultTestId["COMBOBOX_OPTION"] = "combobox-option";
    ComponentDefaultTestId["COLOR_PICKER"] = "color-picker";
    ComponentDefaultTestId["CHECKBOX"] = "checkbox";
    ComponentDefaultTestId["CHECKBOX_LABEL"] = "checkbox-label";
    ComponentDefaultTestId["CHECKBOX_CHECKBOX"] = "checkbox-checkbox";
    ComponentDefaultTestId["DROPDOWN"] = "dropdown";
    ComponentDefaultTestId["DROPDOWN_OPTION_CONTENT"] = "dropdown-option-content";
    ComponentDefaultTestId["BUTTON"] = "button";
    ComponentDefaultTestId["BUTTON_GROUP"] = "button-group";
    ComponentDefaultTestId["CLICKABLE"] = "clickable";
    ComponentDefaultTestId["VIRTUALIZED_LIST"] = "virtualized-list";
    ComponentDefaultTestId["VIRTUALIZED_GRID"] = "virtualized-grid";
    ComponentDefaultTestId["TEXT_FIELD"] = "text-field";
    ComponentDefaultTestId["TEXT_FIELD_SECONDARY_BUTTON"] = "text-field-secondary-button";
    ComponentDefaultTestId["SEARCH"] = "search";
    ComponentDefaultTestId["CLEAN_SEARCH_BUTTON"] = "clean-search-button";
    ComponentDefaultTestId["COLOR_PICKER_ITEM"] = "color-picker-item";
    ComponentDefaultTestId["ICON_BUTTON"] = "icon-button";
    ComponentDefaultTestId["SVG_ICON"] = "svg-icon";
    ComponentDefaultTestId["CHIP"] = "chip";
    ComponentDefaultTestId["RADIO_BUTTON"] = "radio-button";
    ComponentDefaultTestId["RADIO_BUTTON_LABEL"] = "radio-button-label";
    ComponentDefaultTestId["RADIO_BUTTON_CONTROL"] = "radio-button-control";
    ComponentDefaultTestId["TAB"] = "tab";
    ComponentDefaultTestId["TAB_PANEL"] = "tab-panel";
    ComponentDefaultTestId["TAB_PANELS"] = "tab-panels";
    ComponentDefaultTestId["TAB_LIST"] = "tab-list";
    ComponentDefaultTestId["TABS_CONTEXT"] = "tabs-context";
    ComponentDefaultTestId["ALERT_BANNER"] = "alert-banner";
    ComponentDefaultTestId["ALERT_BANNER_BUTTON"] = "alert-banner-button";
    ComponentDefaultTestId["ALERT_BANNER_LINK"] = "alert-banner-link";
    ComponentDefaultTestId["ALERT_BANNER_TEXT"] = "alert-banner-text";
    ComponentDefaultTestId["ATTENTION_BOX"] = "attention-box";
    ComponentDefaultTestId["BOX"] = "box";
    ComponentDefaultTestId["AVATAR"] = "avatar";
    ComponentDefaultTestId["AVATAR_BADGE"] = "avatar-badge";
    ComponentDefaultTestId["AVATAR_CONTENT"] = "avatar-content";
    ComponentDefaultTestId["BREADCRUMB_ITEM"] = "breadcrumb-item";
    ComponentDefaultTestId["BREADCRUMBS_BAR"] = "breadcrumbs-bar";
    ComponentDefaultTestId["BREADCRUMB_MENU"] = "breadcrumb-menu";
    ComponentDefaultTestId["BREADCRUMB_MENU_ITEM"] = "breadcrumb-menu-item";
    ComponentDefaultTestId["PROGRESS_BAR"] = "progress-bar";
    ComponentDefaultTestId["BAR"] = "bar";
    ComponentDefaultTestId["BAR_PRIMARY"] = "bar-primary";
    ComponentDefaultTestId["BAR_SECONDARY"] = "bar-secondary";
    ComponentDefaultTestId["COLOR_PICKER_ITEM_COMPONENT"] = "color-picker-item-component";
    ComponentDefaultTestId["COUNTER"] = "counter";
    ComponentDefaultTestId["MENU"] = "menu";
    ComponentDefaultTestId["OPTION"] = "option";
    ComponentDefaultTestId["EXPAND_COLLAPSE"] = "expand-collapse";
    ComponentDefaultTestId["EDITABLE_INPUT"] = "editable-input";
    ComponentDefaultTestId["EDITABLE_HEADING"] = "editable-heading";
    ComponentDefaultTestId["HEADING"] = "heading";
    ComponentDefaultTestId["LABEL"] = "label";
    ComponentDefaultTestId["LINK"] = "link";
    ComponentDefaultTestId["MENU_DIVIDER"] = "menu-divider";
    ComponentDefaultTestId["MENU_ITEM"] = "menu-item";
    ComponentDefaultTestId["MENU_ITEM_BUTTON"] = "menu-item-button";
    ComponentDefaultTestId["MENU_TITLE"] = "menu-title";
    ComponentDefaultTestId["MENU_TITLE_CAPTION"] = "menu-title-caption";
    ComponentDefaultTestId["MENU_GRID_ITEM"] = "menu-grid-item";
    ComponentDefaultTestId["MENU_BUTTON"] = "menu-button";
    ComponentDefaultTestId["STEP_INDICATOR"] = "step-indicator";
    ComponentDefaultTestId["STEPS"] = "steps";
    ComponentDefaultTestId["STEPS_FORWARD_COMMAND"] = "steps-forward-command";
    ComponentDefaultTestId["STEPS_BACKWARD_COMMAND"] = "steps-backward-command";
    ComponentDefaultTestId["MULTI_STEP_INDICATOR"] = "multi-step-indicator";
    ComponentDefaultTestId["SKELETON"] = "skeleton";
    ComponentDefaultTestId["SPLIT_BUTTON"] = "split-button";
    ComponentDefaultTestId["SPLIT_BUTTON_PRIMARY_BUTTON"] = "split-button-primary-button";
    ComponentDefaultTestId["SPLIT_BUTTON_SECONDARY_BUTTON"] = "split-button-secondary-button";
    ComponentDefaultTestId["TEXT_WITH_HIGHLIGHT"] = "text-with-highlight";
    ComponentDefaultTestId["TOAST"] = "toast";
    ComponentDefaultTestId["TOAST_CONTENT"] = "toast-content";
    ComponentDefaultTestId["TOAST_LINK"] = "toast-link";
    ComponentDefaultTestId["TOAST_BUTTON"] = "toast-button";
    ComponentDefaultTestId["TOAST_CLOSE_BUTTON"] = "toast-close-button";
    ComponentDefaultTestId["TOGGLE"] = "toggle";
    ComponentDefaultTestId["TIPSEEN"] = "tipseen";
    ComponentDefaultTestId["TIPSEEN_CONTENT"] = "tipseen-content";
    ComponentDefaultTestId["TIPSEEN_CONTENT_SUBMIT"] = "tipseen-content-submit";
    ComponentDefaultTestId["TIPSEEN_CONTENT_DISMISS"] = "tipseen-content-dismiss";
    ComponentDefaultTestId["TIPSEEN_TITLE"] = "tipseen-title";
    ComponentDefaultTestId["DIVIDER"] = "divider";
    ComponentDefaultTestId["DATEPICKER"] = "date-picker";
    ComponentDefaultTestId["DATEPICKER_HEADER"] = "date-picker-header";
    ComponentDefaultTestId["DATEPICKER_YEAR_SELECTION"] = "date-picker-year-selection";
    ComponentDefaultTestId["LOADER"] = "loader";
    ComponentDefaultTestId["ICON"] = "icon";
    ComponentDefaultTestId["RESPONSIVE_LIST"] = "responsive-list";
    ComponentDefaultTestId["LIST"] = "list";
    ComponentDefaultTestId["MODAL"] = "modal";
    ComponentDefaultTestId["MODAL_OVERLAY"] = "modal-overlay";
    ComponentDefaultTestId["MODAL_HEADER"] = "modal-header";
    ComponentDefaultTestId["MODAL_CLOSE_BUTTON"] = "modal-close-button";
    ComponentDefaultTestId["MODAL_CONTENT"] = "modal-content";
    ComponentDefaultTestId["MODAL_FOOTER"] = "modal-footer";
    ComponentDefaultTestId["MODAL_MEDIA"] = "modal-media";
    ComponentDefaultTestId["MODAL_BASIC_LAYOUT"] = "modal-basic-layout";
    ComponentDefaultTestId["MODAL_SIDE_BY_SIDE_LAYOUT"] = "modal-side-by-side-layout";
    ComponentDefaultTestId["MODAL_MEDIA_LAYOUT"] = "modal-media-layout";
    ComponentDefaultTestId["FORMATTED_NUMBER"] = "formatted-number";
    ComponentDefaultTestId["HIDDEN_TEXT"] = "hidden-text";
    ComponentDefaultTestId["DIALOG_CONTENT_CONTAINER"] = "dialog-content-container";
    ComponentDefaultTestId["FLEX"] = "flex";
    ComponentDefaultTestId["TOOLTIP"] = "tooltip";
    ComponentDefaultTestId["DIALOG"] = "dialog";
    ComponentDefaultTestId["TABLE"] = "table";
    ComponentDefaultTestId["TABLE_CONTAINER"] = "table-container";
    ComponentDefaultTestId["TABLE_BODY"] = "table-body";
    ComponentDefaultTestId["TABLE_VIRTUALIZED_BODY"] = "table-virtualized-body";
    ComponentDefaultTestId["TABLE_CELL"] = "table-cell";
    ComponentDefaultTestId["TABLE_HEADER"] = "table-header";
    ComponentDefaultTestId["TABLE_HEADER_CELL"] = "table-header-cell";
    ComponentDefaultTestId["TABLE_ROW"] = "table-row";
    ComponentDefaultTestId["TABLE_ROW_MENU"] = "table-row-menu";
})(ComponentDefaultTestId || (ComponentDefaultTestId = {}));
export var NavigationCommand;
(function (NavigationCommand) {
    NavigationCommand["RIGHT_ARROW"] = "{arrowright}";
    NavigationCommand["LEFT_ARROW"] = "{arrowleft}";
    NavigationCommand["UP_ARROW"] = "{arrowup}";
    NavigationCommand["DOWN_ARROW"] = "{arrowdown}";
    NavigationCommand["TAB"] = "#TAB#";
    NavigationCommand["ENTER"] = "{enter}";
    NavigationCommand["PAGE_UP"] = "{pageup}";
    NavigationCommand["PAGE_DOWN"] = "{pagedown}";
})(NavigationCommand || (NavigationCommand = {}));
export var ComponentVibeId;
(function (ComponentVibeId) {
    ComponentVibeId["ACCORDION"] = "Accordion";
    ComponentVibeId["ALERT_BANNER"] = "AlertBanner";
    ComponentVibeId["ATTENTION_BOX"] = "AttentionBox";
    ComponentVibeId["AVATAR"] = "Avatar";
    ComponentVibeId["AVATAR_GROUP"] = "AvatarGroup";
    ComponentVibeId["BADGE"] = "Badge";
    ComponentVibeId["BOX"] = "Box";
    ComponentVibeId["BREADCRUMBS_BAR"] = "BreadcrumbsBar";
    ComponentVibeId["BUTTON"] = "Button";
    ComponentVibeId["BUTTON_GROUP"] = "ButtonGroup";
    ComponentVibeId["CHECKBOX"] = "Checkbox";
    ComponentVibeId["CHIPS"] = "Chips";
    ComponentVibeId["COLOR_PICKER"] = "ColorPicker";
    ComponentVibeId["COMBOBOX"] = "Combobox";
    ComponentVibeId["COUNTER"] = "Counter";
    ComponentVibeId["DATE_PICKER"] = "DatePicker";
    ComponentVibeId["DIALOG"] = "Dialog";
    ComponentVibeId["DIVIDER"] = "Divider";
    ComponentVibeId["DROPDOWN"] = "Dropdown";
    ComponentVibeId["EDITABLE_HEADING"] = "EditableHeading";
    ComponentVibeId["EDITABLE_TEXT"] = "EditableText";
    ComponentVibeId["EXPAND_COLLAPSE"] = "ExpandCollapse";
    ComponentVibeId["FLEX"] = "Flex";
    ComponentVibeId["HEADING"] = "Heading";
    ComponentVibeId["ICON"] = "Icon";
    ComponentVibeId["ICON_BUTTON"] = "IconButton";
    ComponentVibeId["LABEL"] = "Label";
    ComponentVibeId["PROGRESS_BAR"] = "ProgressBar";
    ComponentVibeId["LINK"] = "Link";
    ComponentVibeId["LIST"] = "List";
    ComponentVibeId["LOADER"] = "Loader";
    ComponentVibeId["MENU"] = "Menu";
    ComponentVibeId["MENU_BUTTON"] = "MenuButton";
    ComponentVibeId["MODAL"] = "Modal";
    ComponentVibeId["MULTI_STEP_INDICATOR"] = "MultiStepIndicator";
    ComponentVibeId["RADIO_BUTTON"] = "RadioButton";
    ComponentVibeId["SEARCH"] = "Search";
    ComponentVibeId["SKELETON"] = "Skeleton";
    ComponentVibeId["SLIDER"] = "Slider";
    ComponentVibeId["SPLIT_BUTTON"] = "SplitButton";
    ComponentVibeId["STEPS"] = "Steps";
    ComponentVibeId["TAB"] = "Tab";
    ComponentVibeId["TABLE"] = "Table";
    ComponentVibeId["TEXT"] = "Text";
    ComponentVibeId["TEXT_AREA"] = "TextArea";
    ComponentVibeId["TEXT_FIELD"] = "TextField";
    ComponentVibeId["TEXT_WITH_HIGHLIGHT"] = "TextWithHighlight";
    ComponentVibeId["THEME_PROVIDER"] = "ThemeProvider";
    ComponentVibeId["TIPSEEN"] = "Tipseen";
    ComponentVibeId["TOAST"] = "Toast";
    ComponentVibeId["TOGGLE"] = "Toggle";
    ComponentVibeId["TOOLTIP"] = "Tooltip";
    ComponentVibeId["VIRTUALIZED_GRID"] = "VirtualizedGrid";
    ComponentVibeId["VIRTUALIZED_LIST"] = "VirtualizedList";
})(ComponentVibeId || (ComponentVibeId = {}));
