// @vibe/shared/dist/hooks/useKeyEvent.js
import { useCallback, useRef } from "react";
import { useEventListener } from "./useEventListener";
import { isClient } from "../utils/ssr-utils";
const CTRL_OR_META = "ctrlOrMetaKey";
var Modifier;
(function (Modifier) {
    Modifier["ALT"] = "altKey";
    Modifier["META"] = "metaKey";
    Modifier["CTRL"] = "ctrlKey";
    Modifier["SHIFT"] = "shiftKey";
    Modifier["CTRL_OR_META"] = "ctrlOrMetaKey";
})(Modifier || (Modifier = {}));
const checkModifierInEvent = (event, modifier) => {
    if (modifier === Modifier.CTRL_OR_META) {
        return event.ctrlKey || event.metaKey;
    }
    return event[modifier];
};
const checkWithoutModifierInEvent = (event) => {
    return !Object.values(useKeyEvent.modifiers).some((m) => {
        if (m !== CTRL_OR_META) {
            return !!event[m];
        }
    });
};
export function useKeyEvent({ keys = [], callback, modifier, withoutAnyModifier, ref, ignoreDocumentFallback = false, capture = false, preventDefault = false, stopPropagation = false, keyEventName = "keydown" // need keydown and not keyup to prevent scrolling with prevent default, for example during menu keyboard navigation
 }) {
    const documentRef = useRef(isClient() ? document.body : null);
    const onKeyUpPress = useCallback((event) => {
        const { key } = event;
        if (!keys.includes(key)) {
            return;
        }
        if (modifier && !checkModifierInEvent(event, modifier)) {
            return;
        }
        if (withoutAnyModifier && !checkWithoutModifierInEvent(event)) {
            return;
        }
        if (preventDefault) {
            event.preventDefault();
        }
        if (stopPropagation) {
            event.stopPropagation();
        }
        callback(event);
    }, [keys, modifier, withoutAnyModifier, preventDefault, stopPropagation, callback]);
    let listenerRef;
    if (ref) {
        listenerRef = ref;
    }
    else if (ignoreDocumentFallback) {
        listenerRef = null;
    }
    else {
        listenerRef = documentRef;
    }
    useEventListener({
        eventName: keyEventName,
        callback: onKeyUpPress,
        ref: listenerRef,
        capture
    });
}
useKeyEvent.modifiers = Modifier;
